libraries/AP_Proximity/AP_Proximity_ARS408_CAN.cpp

```diff
@@ -0,0 +1,388 @@
+    return false;
+  }
+  radar_cfg_msg.data.RadarCfg_OutputType = static_cast<uint64_t>(output_type);
+  radar_cfg_msg.data.RadarCfg_OutputType_valid = static_cast<uint64_t>(valid);
+  return true;
+}
+
+void AP_Proximity_ARS408_CAN::set_send_quality(bool quality, bool valid) {
+  radar_cfg_msg.data.RadarCfg_SendQuality = static_cast<uint64_t>(quality);
+  radar_cfg_msg.data.RadarCfg_SendQuality_valid = static_cast<uint64_t>(valid);
+}
+
+void AP_Proximity_ARS408_CAN::set_send_ext_info(bool send_ext, bool valid) {
+  radar_cfg_msg.data.RadarCfg_SendExtInfo = static_cast<uint64_t>(send_ext);
+  radar_cfg_msg.data.RadarCfg_SendExtInfo_valid = static_cast<uint64_t>(valid);
+}
+
+bool AP_Proximity_ARS408_CAN::set_sort_index(int sort_index, bool valid) {
+  if (sort_index < 0 || sort_index > 2) {
+    return false;
+  }
+  radar_cfg_msg.data.RadarCfg_SortIndex = static_cast<uint64_t>(sort_index);
+  radar_cfg_msg.data.RadarCfg_SortIndex_valid = static_cast<uint64_t>(valid);
+  return true;
+}
+
+void AP_Proximity_ARS408_CAN::set_ctrl_relay_cfg(bool ctrl_relay, bool valid) {
+  radar_cfg_msg.data.RadarCfg_CtrlRelay = static_cast<uint64_t>(ctrl_relay);
+  radar_cfg_msg.data.RadarCfg_CtrlRelay_valid = static_cast<uint64_t>(valid);
+}
+
+void AP_Proximity_ARS408_CAN::set_store_in_nvm(bool store_in_nvm, bool valid) {
+  radar_cfg_msg.data.RadarCfg_StoreInNVM = static_cast<uint64_t>(store_in_nvm);
+  radar_cfg_msg.data.RadarCfg_StoreInNVM_valid = static_cast<uint64_t>(valid);
+}
+
+bool AP_Proximity_ARS408_CAN::set_rcs_threshold(int rcs_threshold, bool valid) {
+  if (rcs_threshold != 0 && rcs_threshold != 1) {
+    return false;
+  }
+  radar_cfg_msg.data.RadarCfg_RCS_Threshold = static_cast<uint64_t>(rcs_threshold);
+  radar_cfg_msg.data.RadarCfg_RCS_Threshold_valid = static_cast<uint64_t>(valid);
+  return true;
+}
+
+
+
+// get maximum distance (in meters) of sensor
+float AP_Proximity_ARS408_CAN::distance_max() const
+{
+    return 200.0f;  //200m max range
+}
+
+// get minimum distance (in meters) of sensor
+float AP_Proximity_ARS408_CAN::distance_min() const
+{
+    return 0.5f;  // 0.5 min range 
+}
+
+// handler for incoming frames
+void AP_Proximity_ARS408_CAN::handle_frame(AP_HAL::CANFrame &frame)
+{
+    uint32_t msg_id = frame.id;
+    switch(CAN_ID(msg_id)){
+        // 0x60A
+        case CAN_ID::Object_Status:
+        {
+             _last_reading_ms = AP_HAL::millis();
+             memcpy(obj_status_.raw_data, frame.data, frame.dlc);
+             if(obj_status_.data.Object_NofObjects == 0){
+                set_status(AP_Proximity::Status::NoData);
+             }
+
+            // clear contian list
+            obj_general_list_.clear();
+            obj_quality_list_.clear();
+            obj_extended_list_.clear();
+
+            break;
+        }
+
+        // 0x60B
+        case CAN_ID::Object_General:
+        {
+            object_1_general obj1_general_;
+            memcpy(obj1_general_.raw_data,frame.data, frame.dlc);
```

Comment:
If the radar screws up and sets dlc to a very large number your Plane/Copter might fall out of the sky with this construct due to a memory overwrite.

Can this be done with simple assignment rather than a `memcpy`?